src/main.cpp

```diff
@@ -80,111 +118,110 @@ void setGainsFromRcTrimmers()
  */
 void loop()
 {  
-  
+  taskENTER_CRITICAL_ISR(&timerMux);
+  timerState = !timerState;
+  taskEXIT_CRITICAL_ISR(&timerMux);
+
   float quaternionImuData[4];
   float measuredAngularVelocity[3];
   float angularVelocityCorrected[3];
 
-  // if raspberryEsp32Interface data received update the last data read time.
-  if (raspberryEsp32Interface.readPiData())
+  if (timerState==true)
```

Comment:
To avoid this lengthy if-block you can write 

```c++
if (!timerState) return;
```

Reply:
Sounds good, does it matter if its at start or end?

Reply:
It should be at the start